pkgs/development/python-modules/gpy/default.nix

```diff
@@ -0,0 +1,29 @@
+{ stdenv, buildPythonPackage, fetchPypi
+, numpy, scipy, six, paramz, nose, matplotlib, cython }:
+
+buildPythonPackage rec {
+  pname = "GPy";
+  version = "1.7.7";
+  name  = "${pname}-${version}";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1b4siirlkqic1lsn9bi9mnp8fpbpw1ijwv0z2i6r2zdrk3d6szs1";
+  };
+
+  # running tests produces "ImportError: cannot import name 'linalg_cython'"
+  # even though Cython has run
+  checkPhase = "nosetests -d";
+  doCheck = false;
+
+  buildInputs = [ nose cython matplotlib ];
+
+  propagatedBuildInputs = [ numpy scipy six paramz ];
+
+  meta = with stdenv.lib; {
+    description = "Gaussian process framework in Python";
+    homepage = https://sheffieldml.github.io/GPy;
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ bcdarwin ];
```

Comment:
Why? Do you want to exclude a certain platform? `buildPythonPackage` inherits `platforms` from `python`.